src/BugYield/BugTracker/JiraBugTracker.php

```diff
@@ -25,41 +25,35 @@ public function getUrlTicketPattern() {
     return $this->urlTicketPattern;
   }
 
-  // This shoudn't be called "get", since it's a setter-method.
   public function getApi($url, $username, $password) {
     $this->currentUsername = $username;
-    $this->api= new \SoapClient($url . '/rpc/soap/jirasoapservice-v2?wsdl');
-    $this->token = $this->api->login($username, $password);
+    $this->api = new JiraApi(rtrim($url, '/') . '/rest/api/2/', $username, $password);
   }
 
   /**
-   * Check value of config setting "closed_issue_editable".
-   * If true can update closed jira tickets with worklogs without reopening the tickets.
+   * Get issue based on ticketId.
+   *
+   * @param $ticketId
+   * @return array
    */
-  public function getClosedIssueEditable() {
-    if(isset($this->bugtrackerConfig['closed_issue_editable'])) {
-      if($this->bugtrackerConfig['closed_issue_editable'] === true) {
-        return true;
-      }
-    }
-    return false;
+  public function getIssue($ticketId) {
+    $ticketId = ltrim($ticketId, '#');
+    return $this->api->get($ticketId)->json();
   }
 
   public function getTitle($ticketId) {
-    $ticketId = ltrim($ticketId, '#');
-
     // the Jira throws an exception if the issue does not exists or are unreachable. We don't want that, hence the try/catch
     try {
-      $response = $this->api->getIssue($this->token, $ticketId);
+      $response = $this->getIssue($ticketId);
     } catch (\Exception $e) {
-      // Valuable information will be returned from Jira here, e.g.:
+      //TODO: Valuable information will be returned from Jira here, we should log it somewhere. E.g.:
```

Comment:
Looking at Git Blame (cause I don't really remember writing it tbh), its origin was from when I copied the original SOAP jira bug tracker. I also always comment TODO's with a @ (@TODO), so I'm pretty sure this isn't my work. To make it short: I don't actually know the reason why the comment is relevant